crates/matrix-sdk/src/room/common.rs

```diff
@@ -175,14 +179,19 @@ impl Common {
             state: http_response.state,
         };
 
-        for event in &http_response.chunk {
-            let event = event.deserialize()?;
-
+        for event in http_response.chunk {
             #[cfg(feature = "encryption")]
-            let event = self.client.decrypt_room_event(&event).await?;
+            let event = match event.deserialize() {
+                Ok(event) => self.client.decrypt_room_event(&event).await?,
```

Comment:
Won't a decryption failure here also prevent you from getting any events at all? Not being able to decrypt room history is a common thing.

Reply:
Looking at `decrypt_room_event`, it will only return an error if the decrypted event cannot be deserialized to a json value (I guess this might happen?) serialization of the room_id or the conversion back into raw event fails (very unlikely I presume??). So for the former case it probably makes sense to catch that, yes.

Reply:
I've added another commit to address this, but I'm not 100% happy with it: If decryption is successful, but the decrypted event is not valid json, the undecrypted event is returned by messages.
Another option would be to instead change `decrypt_room_event`: Remove the error cases by not adding the `room_id` field if the message is not valid json (so it cannot be deserialized anyway). Then the user can discover that the inner message is malformed when they try to deserialize it as a `RoomEvent`.
What do you think?

Reply:
Looking at this a bit better, the whole `room_id` insertion dance there is pretty pointless, we already have a `room_id` there. The `room_id` is part of the ciphertext[[1]] and after decryption it will be part of the event[[2]].

I think we can just remove the room id insertion.

Could you also rebase so CI doesn't fail?

[1]: https://spec.matrix.org/v1.1/client-server-api/#mmegolmv1aes-sha2
[2]: https://github.com/matrix-org/matrix-rust-sdk/blob/main/crates/matrix-sdk-crypto/src/olm/group_sessions/inbound.rs#L372

Reply:
So, simply removing the insertion did not work, because the decrypt methods returned a `AnySyncRoomEvent` rather than a `AnyRoomEvent` (or similar). However, since the `room_id` is actually present in the json returned from these methods (since otherwise an error is returned as you've pointed out), I've changed those codepaths to return a `Raw<AnyRoomEvent>` or `RoomEvent` now. In some places this made things simpler (because we could remove the superfluous insertion of the room_id). For the case that a `RoomEvent` is actually needed, I added a noop (at least with optimizations) `From` implementation which in effect just discards the information that a `room_id` field is present.

I've also rebased as requested.